src/components/DocsFooter.tsx

```diff
@@ -78,12 +78,12 @@ function FooterLink({
         className="text-tertiary dark:text-tertiary-dark inline group-focus:text-link dark:group-focus:text-link-dark"
         displayDirection={type === 'Previous' ? 'start' : 'end'}
       />
-      <span>
-        <span className="block no-underline text-sm tracking-wide text-secondary dark:text-secondary-dark uppercase font-bold group-focus:text-link dark:group-focus:text-link-dark group-focus:text-opacity-100">
+      <div className="flex flex-col overflow-hidden">
```

Comment:
I would probably update the tailwind config in this PR and do the tailwind upgrade in a separate PR.

In `tailwind.config.js` replace:

```js
maxWidth: {
  xs: '21rem',
},
```

With:

```js
maxWidth: {
  // add this because it should be there anyway
  ...defaultTheme.maxWidth, 
  xs: '21rem',
},
minWidth: {
  ...defaultTheme.minWidth,
  80: '20rem',
},
```

Reply:
Hey @rickhanlonii thank you for reviewing my PR. I agree with your suggestions, but I feel there's a small caveat. If we set the width to `w-fit` on the `NextLink` directly, there'd be no upper limit, meaning extra long strings would overlap the adjacent **PREVIOUS** card.
![case1](https://github.com/reactjs/react.dev/assets/22348265/8daa0340-bc36-43da-88cd-ead4f9b1c5ee)
I could think of 2 ways to solve this:
**METHOD 1 - Remove explicit width**
Instead we could remove the explicit width on the `NextLink`  and add only `md:w-fit` to the child `div` (since there's no width defined on the parent wrapper element, each hoverable card would occupy the full space, i.e, 50% of the total space available) regardless of the size of the content rendered within. 
![case5](https://github.com/reactjs/react.dev/assets/22348265/a67dc57b-222a-4d89-bef1-ab2c6b318a09)

![case2](https://github.com/reactjs/react.dev/assets/22348265/193a8451-b0ac-45da-b3b0-ac2de291f5c7)

However, longer strings are rendered on multiple lines, so we retain the `overflow-hidden`.
![case3](https://github.com/reactjs/react.dev/assets/22348265/e9d4a196-ee2a-42a0-8495-724b707dfc48)


**METHOD 2 - Set max width along with min width**
We add `min-w-80` and `w-fit` and along with it we also set the max-width (`max-w-md`), and retain the overflow property to handle the string that is longer than what the max width can accommodate. 

https://github.com/reactjs/react.dev/assets/22348265/ab789c40-f966-41fb-af9e-9dcbd98d5d3d

Currently, I have implemented the second approach. But I'd love to know your thoughts on this and if I should change anything further.

Reply:
Good point. I'm not concerned about it for APIs, but for translations this could be a problem. 

Can method 2 wrap at max width instead of truncating?

Reply:
Yes, I think that's a good idea. I tried adding a long string to the footer link with wrap instead of truncating it, and, we're able to show the complete string without loss of information(non-truncating). It is also in-line with how long strings are rendered on the sidebar.
![wrap2](https://github.com/reactjs/react.dev/assets/22348265/dcd85453-9fb6-4243-b14f-c3369c7791d4)

I've implemented `break-words` to `break-normal`  for wrapping the string.
![wrap](https://github.com/reactjs/react.dev/assets/22348265/4de46224-77e9-4a63-b262-7670e877fa76)